### DISCUSSION TITLE
Inconsistent results when using 'driver.find_element_by_accessibility_id'

### DISCUSSION THREAD

**Discussion by shaky** [appium/appium#15692] [Q&A]
hello community,

I recently switched from appium 1.18.3 + Xcode 12.1 to appium 1.21.0 + Xcode 12.5 and I noticed that some of my tests are failing, mainly because of the driver.find_element_by_accessibility_id behaviour.

Here's the current behaviour I see(with Xcode 12.1 to appium 1.21.0):

| | self.driver.find_element_by_accessibility_id('Button Text') | self.driver.find_element_by_accessibility_id('Label') | self.driver.find_element_by_accessibility_id('Identifier') |
| -- | --- | --- | -- |
| Button Text | working | N/A| N/A |
| Button Text + Label| working| working| N/A |
| Button Text + Identifier| working| N/A| working |
| Button text + Label + Identifier| working| not working| working |

|    | self.driver.find_element_by_accessibility_id('Label Text')  | self.driver.find_element_by_accessibility_id('Label') | self.driver.find_element_by_accessibility_id('Identifier') |
| -- | --- | --- | -- |
| Label Text | working | N/A | N/A |
| Label Text + Label | not working | working | N/A |
| Button Text + Identifier | not working | N/A | working |
| Label text + Label + Identifier | not working | not working | working |

My questions are:
1. For the scenario where label and identifier are both defined, is it expected that we can't find and element by its Label anymore?
2. Why the behaviour changed between the updates?
3. Is there a best practice on how to use 'find_element_by_accessibility_id' ?

Thank you!

**Comment by KazuCocoa** [comment]
Did you check a breaking change in 1.20.0? https://github.com/appium/appium/blob/master/CHANGELOG.md#iosxcuitest-1

Priority of accessibilityIdentifier, accessibilityLabel,  and other accessibility stuff ( https://developer.apple.com/documentation/uikit/uiaccessibilityidentification/1623132-accessibilityidentifier , https://developer.apple.com/documentation/uikit/uiaccessibilityelement ) depend on how iOS handle them.
accessibilityIdentifier is top priority in their internal. (I cannot find this documentation, but Apple published this priority long before)

Personally, it would be good to use `find_element_by_accessibility_id` for accessibilityIdentifier (primary) and text.
Others are via predicate strategy.
